### PR TITLE
Migrate developer docs for DogStatsD internals

### PR DESCRIPTION
**Pull request by ofek**
### What does this PR do?

This migrates just: https://github.com/DataDog/datadog-agent/blob/main/docs/dogstatsd/internals.md

The rest are customer facing and are already on the main website.

### CODE REVIEW

**Review by chouetz — APPROVED** [review]

**Review by estherk15 — APPROVED** [review]
No docs review needed

File: docs/public/architecture/dogstatsd/internals.md [hunk 1]
```diff
@@ -1,100 +1,92 @@
-# DogStatsD
+# DogStatsD internals
 
-Information on DogStatsD, configuration and troubleshooting is available in the Datagdog documentation:
-[https://docs.datadoghq.com/developers/dogstatsd/][1]
```
Comment (vickenty) [inline docs/public/architecture/dogstatsd/internals.md:4]: Would it make sense to keep a link to the user documentation?

**Review by vickenty — APPROVED** [review]

File: docs/public/architecture/dogstatsd/internals.md (hunk 1 above)
Comment (ofek) [reply docs/public/architecture/dogstatsd/internals.md:4]: Do you think having that link would be useful to us? Customers would not be using with page.

Comment (vickenty) [reply docs/public/architecture/dogstatsd/internals.md:4]: Not to us, but to people that land on this page from internet search. There is no indication on the page that these are internal docs not intended for end users.

Comment (ofek) [reply docs/public/architecture/dogstatsd/internals.md:4]: Added!

**Pull request merged**